### PR TITLE
Split capabilities package into separate file

### PR DESCRIPTION
**Pull request by zyga**
NOTE: this is based on #128 so please land that first.

This doing a split that everyone has asked for, capabilities.go becomes {capabilities,repo,misc}.go. Same thing is applied to tests.

### CODE REVIEW

**Comment by zyga** [comment]
Fixed go fmt and re-pushed. Sorry for the noise.

**Comment by niemeyer** [comment]
LGTM

File: caps/misc.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// ValidateName checks if a string as a capability name
+func ValidateName(name string) error {
+	valid := validName.MatchString(name)
+	if !valid {
+		return fmt.Errorf("%q is not a valid snap name", name)
+	}
+	return nil
+}
+
+func (e *NotFoundError) Error() string {
```
Comment (chipaca) [inline caps/misc.go]: any reason this is separated from the type definition by ValidateName?

File: caps/misc.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	case "remove":
+		return fmt.Sprintf("can't remove capability %q, no such capability", e.name)
+	default:
+		panic(fmt.Sprintf("unexpected what: %q", e.what))
+	}
+}
+
+// LoadBuiltInTypes adds all built-in types to the repository
+// If any of the additions fail the function returns the error and stops.
+func LoadBuiltInTypes(r *Repository) error {
```
Comment (chipaca) [inline caps/misc.go]: any reason this isn't a method of Repository?

File: caps/repo.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+		if t == otherT {
+			return fmt.Errorf("cannot add type %q: name already exists", t)
+		}
+	}
+	r.types = append(r.types, t)
+	return nil
+}
+
+// Remove removes the capability with the provided name.
+// Removing a capability that doesn't exist silently does nothing
```
Comment (chipaca) [inline caps/repo.go]: returning an error isn't "silently", though

File: caps/types.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+var builtInTypes = [...]Type{
+	FileType,
+}
+
+// String returns a string representation for the capability type.
+func (t Type) String() string {
+	return string(t)
+}
+
+// Validate if a capability is correct according to the given type
```
Comment (chipaca) [inline caps/types.go]: whether, not if

**Comment by chipaca** [comment]
silly things i'll let you address down the line. All in all, :+1: .

File: caps/misc.go (hunk 2 above)
Comment (zyga) [inline caps/misc.go]: Gustavo asked it not to be. It was originally.

File: caps/misc.go (hunk 1 above)
Comment (zyga) [inline caps/misc.go]: Not really, just copy-paste around. I can correct that.

File: caps/repo.go (hunk 3 above)
Comment (zyga) [inline caps/repo.go]: Oh, that's true. I must have messed up something (or I was blind and didn't update the docstring). I'll fix it.

File: caps/types.go (hunk 4 above)
Comment (zyga) [inline caps/types.go]: Thanks

**Pull request merged**